codetransformer/code.py

```diff
@@ -0,0 +1,432 @@
+                ),
+            ),
+            self.flags,
+            bytes(bc),
+            consts,
+            names,
+            varnames,
+            self.filename,
+            self.name,
+            self.firstlineno,
+            self.lnotab,
+            freevars,
+            cellvars,
+        )
+
+    @property
+    def instrs(self):
+        return self._instrs
+
+    @property
+    def sparse_instrs(self):
+        return tuple(_sparse_args(self.instrs))
+
+    @property
+    def argcount(self):
+        return self._argcount
+
+    @property
+    def kwonlyargcount(self):
+        return self._kwonlyargcount
+
+    @property
+    def consts(self):
+        # We cannot use a set comprehension because consts do not need
+        # to be hashable.
+        consts = []
+        append_const = consts.append
+        for instr in self.instrs:
+            if isinstance(instr, LOAD_CONST) and instr.arg not in consts:
+                append_const(instr.arg)
+        return tuple(consts)
+
+    @property
+    def names(self):
+        return tuple(sorted({
+            instr.arg for instr in self.instrs if instr.uses_name
+        }))
+
+    @property
+    def argnames(self):
+        return self._argnames
+
+    @property
+    def varnames(self):
+        return self._argnames + tuple(sorted({
+            instr.arg for instr in self.instrs if instr.uses_varname
+        }))
+
+    @property
+    def cellvars(self):
+        return self._cellvars
+
+    @property
+    def freevars(self):
+        return self._freevars
+
+    @property
+    def flags(self):
+        return self._flags
+
+    @property
+    def nested(self):
+        return bool(self._flags & Flags.CO_NESTED)
+
+    @property
+    def generator(self):
+        return bool(self._flags & Flags.CO_GENERATOR)
+
+    @property
+    def coroutine(self):
+        return bool(self._flags & Flags.CO_COROUTINE)
+
+    @property
+    def iterable_coroutine(self):
+        return bool(self._flags & Flags.CO_ITERABLE_COROUTINE)
+
+    @property
+    def filename(self):
+        return self._filename
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def firstlineno(self):
+        return self._firstlineno
+
+    @property
+    def lnotab(self):
+        return self._lnotab
+
+    @property
+    def stack_effect(self):
+        return max(scanl(
+            operator.add,
+            0,
+            map(operator.attrgetter('stack_effect'), self.instrs),
+        ))
+
+    def __getitem__(self, idx):
+        return self.instrs[idx]
+
+    def index(self, instr):
+        """Returns the index of instr.
+
+        Parameters
+        ----------
+        instr : Instruction
+            The instruction the check the index of.
+
+        Returns
+        -------
+        idx : int
+            The index of instr in this code object.
+        """
+        return self._instrs.index(instr)
+
+    def sparse_index(self, instr):
```

Comment:
This feels like an implementation detail.  Should it be public? When would you want this from an external consumer?